conda/cli/install.py

```diff
@@ -479,6 +452,72 @@ def install(args, parser, command="install"):
     handle_txn(unlink_link_transaction, prefix, args, newenv)
 
 
+def install_revision(args, parser):
```

Comment:
Since both `install` and `install_revision` end up with `handle_txn()`, would it make sense to architect this refactor in a way that `install*` functions return the transaction, and this is handled in a layer above?

Reply:
There is also `install_clone` to handle. This currently does not return a transaction, instead it executes the cloning action and prints the activation message. I think this can be refactored to be more in line with `install` and `install_revision`. 

However, I think it makes sense to me for the cli layer to deal with:
* validating that combinations of command line arguments are valid
* doing preliminary checks + validations to ensure that the whatever actions can be successful
* doing any post installation steps/validations/messaging

Then the install layer to deal with pulling together the appropriate plan for installation (given the command line arguments) and executing it.

How does that sound to you?

Reply:
Yes, in general I'd like to have a streamlined way of running all necessary pre-checks, then creating a transaction object, then reporting and post-checks.

Creating the transaction object here is the key step, because not every installation process results in one right now. Clones, explicits, revisions, and regular `conda create|install|update|remove` operations should all result in a transaction. Once we have a transaction, then reporting can be unified.

Reply:
I think that's what I mean by transaction is what you meant by "installation plan", so we are in agreement.

Reply:
Hmmm, ya I think we have a similar vision. I think this PR doesn't get all the way there. There is still more work to be done on aspects like collecting and parsing input files etc. I would advocate for punting on solving that fully for the purpose of this PR, so as not to explode the scope too much (I'd say the scope of this PR is to create a set of reusable validations, clarify the difference between different installation workflows and reduce the responsibility of `conda.cli.install.install`). But certainly address it in follow up work.